djstripe/models.py

```diff
@@ -441,26 +446,28 @@ def sync_charges(self, cu=None, **kwargs):
 
     def sync_current_subscription(self, cu=None):
         cu = cu or self.stripe_customer
-        sub = cu.subscription
+        sub = getattr(cu, 'subscription', None)
+        cur_sub = getattr(self, 'current_subscription', None)
         if sub:
-            try:
-                sub_obj = self.current_subscription
-                sub_obj.plan = djstripe_settings.plan_from_stripe_id(sub.plan.id)
-                sub_obj.current_period_start = convert_tstamp(
+            if cur_sub:
+                _log.debug('Updating subscription')
+                cur_sub.plan = djstripe_settings.plan_from_stripe_id(sub.plan.id)
+                cur_sub.current_period_start = convert_tstamp(
                     sub.current_period_start
                 )
-                sub_obj.current_period_end = convert_tstamp(
+                cur_sub.current_period_end = convert_tstamp(
                     sub.current_period_end
                 )
-                sub_obj.amount = (sub.plan.amount / decimal.Decimal("100"))
-                sub_obj.status = sub.status
-                sub_obj.cancel_at_period_end = sub.cancel_at_period_end
-                sub_obj.canceled_at = convert_tstamp(sub, "canceled_at")
-                sub_obj.start = convert_tstamp(sub.start)
-                sub_obj.quantity = sub.quantity
-                sub_obj.save()
-            except CurrentSubscription.DoesNotExist:
-                sub_obj = CurrentSubscription.objects.create(
+                cur_sub.amount = (sub.plan.amount / decimal.Decimal("100"))
+                cur_sub.status = sub.status
+                cur_sub.cancel_at_period_end = sub.cancel_at_period_end
+                cur_sub.canceled_at = convert_tstamp(sub, "canceled_at")
+                cur_sub.start = convert_tstamp(sub.start)
+                cur_sub.quantity = sub.quantity
+                cur_sub.save()
+            else:
+                _log.debug('Creating subscription')
```

Comment:
It's not good practice since log formatting is used to print out time, module name, function, etc. For example, a log format like this is standard: 
`%(asctime)s [%(levelname)s] %(name)s.%(funcName)s: %(message)s`

You can also use logging config to redirect all djstripe logs to a specific formatter and, for example, highlight them green.